Multisig/Logic/Models/TokenRegistry/TokenRegistry.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  TokenRegistry.swift
+//  Multisig
+//
+//  Created by Dmitry Bespalov on 09.07.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+
+class TokenRegistry {
+
+    private var hardcoded = HardcodedTokenStore()
+    private var cache = InMemoryTokenStore()
+    private var backend = BackendTokenStore()
+    private var blockchain = BlockchainTokenStore()
+
+    // added for thread safety
+    private var queue = DispatchQueue(label: "SerialTokenRegistry")
+
+    func load(completion: @escaping () -> Void = {}) {
+        dispatchPrecondition(condition: .notOnQueue(queue))
```

Comment:
why is this precondition needed?

Reply:
this precondition is needed to prevent potential deadlock (defensive programming)